akka-projection-slick/src/main/scala/akka/projection/slick/internal/SlickOffsetStore.scala

```diff
@@ -50,8 +50,7 @@ import slick.jdbc.JdbcProfile
     results.map {
       case Nil => None
       case reps if reps.forall(_.mergeable) =>
-        Some(
-          fromStorageRepresentation[MergeableOffset[_, _], Offset](MultipleOffsets(reps.toList)).asInstanceOf[Offset])
+        Some(fromStorageRepresentation[MergeableOffset[_], Offset](MultipleOffsets(reps.toList)).asInstanceOf[Offset])
```

Comment:
He is the original bug. 

When we come to that point, `Offset` was `GroupedOffset`. Although it extends `MergeableOffset`, the actual instance is not a `GroupedOffset` and we get the `ClassCastException`

Bottom line is, we can't support extensions of `MergeableOffset` and the key must have a fixed `String` type.